src/main/java/com/exasol/adapter/dialects/snowflake/SnowflakeSqlDialect.java

```diff
@@ -0,0 +1,196 @@
+import com.exasol.adapter.jdbc.*;
+import com.exasol.adapter.sql.ScalarFunction;
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * This class implements the Snowflake dialect.
+ */
+public class SnowflakeSqlDialect extends AbstractSqlDialect {
+    static final String NAME = "SNOWFLAKE";
+
+    private static final Set<ScalarFunctionCapability> DISABLED_SCALAR_FUNCTION = Set.of(
+            /*
+             * Implementation for `BETWEEN` time functions is not supported. For more information see `design.md` file,
+             * `Scalar Functions` section
+             */
+            SECONDS_BETWEEN, MINUTES_BETWEEN, HOURS_BETWEEN, DAYS_BETWEEN, MONTHS_BETWEEN, YEARS_BETWEEN, //
+            ROUND, // Snowflake rounds `0.5` down while Exasol rounds it up
+            SECOND, // It presents precision issues
+            COLOGNE_PHONETIC, // No Snowflake equivalent
+            CONCAT, // It fails for boolean data types
+            INSTR, // not implemented; probably possible using strpos
+            POSIX_TIME, // Does not follow Exasol session timezone
+            // Currently not implemented:
+            DUMP, EDIT_DISTANCE, INSERT, LOCATE, REGEXP_INSTR, REGEXP_SUBSTR, SOUNDEX, SPACE, UNICODE, UNICODECHR,
+            DBTIMEZONE, FROM_POSIX_TIME, HOUR, SESSIONTIMEZONE, IS_NUMBER, IS_BOOLEAN, IS_DATE, IS_DSINTERVAL,
+            IS_YMINTERVAL, IS_TIMESTAMP, TO_CHAR, TO_DATE, TO_NUMBER, TO_TIMESTAMP, BIT_AND, BIT_CHECK, BIT_LROTATE,
+            BIT_LSHIFT, BIT_NOT, BIT_OR, BIT_RROTATE, BIT_RSHIFT, BIT_SET, BIT_TO_NUM, BIT_XOR, HASHTYPE_MD5, HASH_SHA1,
+            HASHTYPE_SHA1, HASH_SHA256, HASHTYPE_SHA256, HASH_SHA512, HASHTYPE_SHA512, HASH_TIGER, HASHTYPE_TIGER,
+            NULLIFZERO, ZEROIFNULL, MIN_SCALE, NUMTOYMINTERVAL, JSON_VALUE, TO_DSINTERVAL, CONVERT_TZ, NUMTODSINTERVAL,
+            TO_YMINTERVAL, CAST, SYS_GUID, SYSTIMESTAMP, CURRENT_STATEMENT, CURRENT_USER, SYSDATE, CURRENT_SESSION, //
+            // Geospatial are currently not supported:
+            ST_X, ST_Y, ST_ENDPOINT, ST_ISCLOSED, ST_ISRING, ST_LENGTH, ST_NUMPOINTS, ST_POINTN, ST_STARTPOINT, ST_AREA,
+            ST_EXTERIORRING, ST_INTERIORRINGN, ST_NUMINTERIORRINGS, ST_GEOMETRYN, ST_NUMGEOMETRIES, ST_BOUNDARY,
+            ST_BUFFER, ST_CENTROID, ST_CONTAINS, ST_CONVEXHULL, ST_CROSSES, ST_DIFFERENCE, ST_DIMENSION, ST_DISJOINT,
+            ST_DISTANCE, ST_ENVELOPE, ST_EQUALS, ST_FORCE2D, ST_GEOMETRYTYPE, ST_INTERSECTION, ST_INTERSECTS,
+            ST_ISEMPTY, ST_ISSIMPLE, ST_OVERLAPS, ST_SETSRID, ST_SYMDIFFERENCE, ST_TOUCHES, ST_TRANSFORM, ST_UNION,
+            ST_WITHIN);
+    private static final Capabilities CAPABILITIES = createCapabilityList();
+    static final String DATABASE_NAME_PROPERTY = "DATABASE_NAME";
+    static final String ACCOUNT_NAME_PROPERTY = "ACCOUNT_NAME";
+
+    /*
+     * IMPORTANT! Before adding new capabilities, check the `doc/design.md` file if there is a note on why the
+     * capability is not supported. it.
```

Comment:
File `design.md` is empty. Please add details.
```suggestion
     * IMPORTANT! Before adding new capabilities, check the `doc/design.md` file if there is a note on why the
     * capability is not supported. 
```